crates/ra_hir/src/nameres.rs

```diff
@@ -135,7 +135,7 @@ impl<T> PerNs<T> {
         PerNs { types: self.types.as_ref(), values: self.values.as_ref() }
     }
 
-    pub fn combine(self, other: PerNs<T>) -> PerNs<T> {
+    pub fn or(self, other: PerNs<T>) -> PerNs<T> {
```

Comment:
This is literally `std::ops::Or`, isn't it? (I was about to say the same for `as_ref` but it see it's internally borrowed)

Reply:
It *could* be `|`, but iirc `std::option::Option` does not implement this, so we shouldn't either.